### PR TITLE
fix(server/main): correctly send no data to client if there is no skin saved in…

### PR DESCRIPTION
**Pull request by mahanmoulaei**
… the database

### CODE REVIEW

**Comment by thelindat** [comment]
> appearance ~= nil

Why?

> correctly send no data

What is this fixing?

File: server/main.lua [hunk 1]
```diff
@@ -56,7 +56,7 @@ if GetResourceState('es_extended'):find('start') then
 		local xPlayer = ESX.GetPlayerFromId(source)
 		local appearance = MySQL.scalar.await('SELECT skin FROM users WHERE identifier = ?', { xPlayer.identifier })
 
-		cb(appearance and json.decode(appearance) or {})
+		cb(appearance ~= nil and json.decode(appearance) or nil)
```
Comment (thelindat) [inline server/main.lua:59]:
```lua
cb(appearance and json.decode(appearance)
```

**Comment by mahanmoulaei** [comment]
Because in client-side when you trigger this callback, you would probably write it like this:

```
ESX.TriggerServerCallback('esx_skin:getPlayerSkin', function(skin, jobSkin)
        if skin then
                --if code
        else
                --else code
        end
end)
```

**Issue:** if you trigger this callback and the player has no skin saved in the database, the code would still read/run the --if code which it shouldn't be our case! 

If the player has no skin saved to the database, the code should run the --else code part instead. The commit I want to merge fixes this issue.

**Comment by thelindat** [comment]
There are resources that do `esx_skin:getPlayerSkin` and immediately `TriggerEvent('skinchanger:loadSkin', skin)`.

image

Such a change would cause errors.

**Comment by mahanmoulaei** [comment]
> There are resources that do `esx_skin:getPlayerSkin` and immediately `TriggerEvent('skinchanger:loadSkin', skin)`.
> 
> image
> 
> Such a change would cause errors.

Such errors could be easily resolved by adding a simple 
```
if skin then
    TriggerEvent('skinchanger:loadSkin', skin)
end
``` 
to the resources that use ```esx_skin:getPlayerSkin```; but I totally understand what you're saying. Close this PR if you don't need it.

**Comment by thelindat** [comment]
There's no point changing it when it's not going to be compatible with many resources, and skin should always be defined after initial registration. Breaking resources that rely on methods like `next(skin)` or `skin.sex` isn't beneficial.

**Pull request closed without merging**